oc export imagestream always fails

When creating a template from existing project using this command:

```
oc export dc,is,se,route --all -o yaml --as-template=hello-php > test-hello-php.yaml
```

We get `panic: assignment to entry in nil map`. The `export imagestream` part fails.

```
$ oc export is --all -o yaml
panic: assignment to entry in nil map

goroutine 1 [running]:
github.com/openshift/origin/pkg/cmd/cli/cmd.(*defaultExporter).Export(0x1e698f0, 0x7fd113597b68, 0xc2083363c0, 0x0, 0x0, 0x0)
	/data/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/pkg/cmd/cli/cmd/exporter.go:177 +0xbae
github.com/openshift/origin/pkg/cmd/cli/cmd.RunExport(0xc2081c9f20, 0x7fd1135a18a0, 0x1e698f0, 0x7fd113595420, 0xc20802e000, 0x7fd113595370, 0xc20802e008, 0xc208270340, 0xc208228780, 0x1, ...)
	/data/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/pkg/cmd/cli/cmd/export.go:122 +0x7a4
github.com/openshift/origin/pkg/cmd/cli/cmd.func·015(0xc208270340, 0xc208228780, 0x1, 0x4)
	/data/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/pkg/cmd/cli/cmd/export.go:62 +0x118
github.com/spf13/cobra.(*Command).execute(0xc208270340, 0xc208228740, 0x4, 0x4, 0x0, 0x0)
	/data/src/github.com/openshift/origin/Godeps/_workspace/src/github.com/spf13/cobra/command.go:495 +0x65c
github.com/spf13/cobra.(*Command).Execute(0xc2080d8820, 0x0, 0x0)
	/data/src/github.com/openshift/origin/Godeps/_workspace/src/github.com/spf13/cobra/command.go:560 +0x18d
main.main()
	/data/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/cmd/oc/oc.go:22 +0x175

goroutine 5 [syscall]:
os/signal.loop()
	/usr/lib/golang/src/os/signal/signal_unix.go:21 +0x1f
created by os/signal.init·1
	/usr/lib/golang/src/os/signal/signal_unix.go:27 +0x35

goroutine 6 [chan receive]:
github.com/golang/glog.(*loggingT).flushDaemon(0x1e5bb20)
	/data/src/github.com/openshift/origin/Godeps/_workspace/src/github.com/golang/glog/glog.go:879 +0x78
created by github.com/golang/glog.init·1
	/data/src/github.com/openshift/origin/Godeps/_workspace/src/github.com/golang/glog/glog.go:410 +0x2a7

goroutine 30 [select]:
net/http.(*persistConn).writeLoop(0xc208188790)
	/usr/lib/golang/src/net/http/transport.go:945 +0x41d
created by net/http.(*Transport).dialConn
	/usr/lib/golang/src/net/http/transport.go:661 +0xcbc

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
	/usr/lib/golang/src/runtime/asm_amd64.s:2232 +0x1

goroutine 27 [IO wait]:
net.(*pollDesc).Wait(0xc2080f7c60, 0x72, 0x0, 0x0)
	/usr/lib/golang/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc2080f7c60, 0x0, 0x0)
	/usr/lib/golang/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).Read(0xc2080f7c00, 0xc208258c00, 0x400, 0x400, 0x0, 0x7fd113595470, 0xc20808cc00)
	/usr/lib/golang/src/net/fd_unix.go:242 +0x40f
net.(*conn).Read(0xc2081da480, 0xc208258c00, 0x400, 0x400, 0x0, 0x0, 0x0)
	/usr/lib/golang/src/net/net.go:121 +0xdc
crypto/tls.(*block).readFromUntil(0xc20822f590, 0x7fd1135a2bf8, 0xc2081da480, 0x5, 0x0, 0x0)
	/usr/lib/golang/src/crypto/tls/conn.go:454 +0xe6
crypto/tls.(*Conn).readRecord(0xc208255080, 0x17, 0x0, 0x0)
	/usr/lib/golang/src/crypto/tls/conn.go:539 +0x2da
crypto/tls.(*Conn).Read(0xc208255080, 0xc208213000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/lib/golang/src/crypto/tls/conn.go:904 +0x166
net/http.noteEOFReader.Read(0x7fd1135a2fb0, 0xc208255080, 0xc208186738, 0xc208213000, 0x1000, 0x1000, 0x10bcf00, 0x0, 0x0)
	/usr/lib/golang/src/net/http/transport.go:1270 +0x6e
net/http.(*noteEOFReader).Read(0xc20825eca0, 0xc208213000, 0x1000, 0x1000, 0xc208012000, 0x0, 0x0)
	<autogenerated>:125 +0xd4
bufio.(*Reader).fill(0xc208246060)
	/usr/lib/golang/src/bufio/bufio.go:97 +0x1ce
bufio.(*Reader).Peek(0xc208246060, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/lib/golang/src/bufio/bufio.go:132 +0xf0
net/http.(*persistConn).readLoop(0xc2081866e0)
	/usr/lib/golang/src/net/http/transport.go:842 +0xa4
created by net/http.(*Transport).dialConn
	/usr/lib/golang/src/net/http/transport.go:660 +0xc9f

goroutine 28 [select]:
net/http.(*persistConn).writeLoop(0xc2081866e0)
	/usr/lib/golang/src/net/http/transport.go:945 +0x41d
created by net/http.(*Transport).dialConn
	/usr/lib/golang/src/net/http/transport.go:661 +0xcbc

goroutine 29 [IO wait]:
net.(*pollDesc).Wait(0xc208302060, 0x72, 0x0, 0x0)
	/usr/lib/golang/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc208302060, 0x0, 0x0)
	/usr/lib/golang/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).Read(0xc208302000, 0xc208318000, 0x800, 0x800, 0x0, 0x7fd113595470, 0xc20808ce00)
	/usr/lib/golang/src/net/fd_unix.go:242 +0x40f
net.(*conn).Read(0xc20830c000, 0xc208318000, 0x800, 0x800, 0x0, 0x0, 0x0)
	/usr/lib/golang/src/net/net.go:121 +0xdc
crypto/tls.(*block).readFromUntil(0xc20821a180, 0x7fd1135a2bf8, 0xc20830c000, 0x5, 0x0, 0x0)
	/usr/lib/golang/src/crypto/tls/conn.go:454 +0xe6
crypto/tls.(*Conn).readRecord(0xc208310000, 0x17, 0x0, 0x0)
	/usr/lib/golang/src/crypto/tls/conn.go:539 +0x2da
crypto/tls.(*Conn).Read(0xc208310000, 0xc208245000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/lib/golang/src/crypto/tls/conn.go:904 +0x166
net/http.noteEOFReader.Read(0x7fd1135a2fb0, 0xc208310000, 0xc2081887e8, 0xc208245000, 0x1000, 0x1000, 0x10bcf00, 0x0, 0x0)
	/usr/lib/golang/src/net/http/transport.go:1270 +0x6e
net/http.(*noteEOFReader).Read(0xc20825ed60, 0xc208245000, 0x1000, 0x1000, 0xc208012000, 0x0, 0x0)
	<autogenerated>:125 +0xd4
bufio.(*Reader).fill(0xc208246180)
	/usr/lib/golang/src/bufio/bufio.go:97 +0x1ce
bufio.(*Reader).Peek(0xc208246180, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/lib/golang/src/bufio/bufio.go:132 +0xf0
net/http.(*persistConn).readLoop(0xc208188790)
	/usr/lib/golang/src/net/http/transport.go:842 +0xa4
created by net/http.(*Transport).dialConn
	/usr/lib/golang/src/net/http/transport.go:660 +0xc9f
```

--- a/pkg/cmd/cli/cmd/exporter.go
+++ b/pkg/cmd/cli/cmd/exporter.go
@@ -163,7 +163,9 @@ func (e *defaultExporter) Export(obj runtime.Object, exact bool) error {
 			if err != nil {
 				return err
 			}
-			newSpec := imageapi.ImageStreamSpec{}
+			newSpec := imageapi.ImageStreamSpec{
+				Tags: map[string]imageapi.TagReference{},
+			}
 			for name, tag := range t.Status.Tags {
 				if len(tag.Items) > 0 {
 					// copy annotations
@@ -185,12 +187,16 @@ func (e *defaultExporter) Export(obj runtime.Object, exact bool) error {
 				newSpec.Tags[name] = ref
 			}
 			t.Spec = newSpec
-			t.Status = imageapi.ImageStreamStatus{}
+			t.Status = imageapi.ImageStreamStatus{
+				Tags: map[string]imageapi.TagEventList{},
+			}
 			break
 		}
 
 		// otherwise, try to snapshot the most recent image as spec items
-		newSpec := imageapi.ImageStreamSpec{}
+		newSpec := imageapi.ImageStreamSpec{
+			Tags: map[string]imageapi.TagReference{},
+		}
 		for name, tag := range t.Status.Tags {
 			if len(tag.Items) > 0 {
 				// copy annotations
@@ -203,7 +209,9 @@ func (e *defaultExporter) Export(obj runtime.Object, exact bool) error {
 			}
 		}
 		t.Spec = newSpec
-		t.Status = imageapi.ImageStreamStatus{}
+		t.Status = imageapi.ImageStreamStatus{
+			Tags: map[string]imageapi.TagEventList{},
+		}
 
 	case *imageapi.ImageStreamTag:
 		exportObjectMeta(&t.Image.ObjectMeta, exact)

--- a/pkg/cmd/cli/cmd/exporter_test.go
+++ b/pkg/cmd/cli/cmd/exporter_test.go
@@ -1,0 +1,96 @@
+package cmd
+
+import (
+	"reflect"
+	"testing"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+
+	deployapi "github.com/openshift/origin/pkg/deploy/api"
+	deploytest "github.com/openshift/origin/pkg/deploy/api/test"
+	imageapi "github.com/openshift/origin/pkg/image/api"
+)
+
+func TestExport(t *testing.T) {
+	exporter := &defaultExporter{}
+
+	tests := []struct {
+		name        string
+		object      runtime.Object
+		exact       bool
+		expectedObj runtime.Object
+		expectedErr error
+	}{
+		{
+			name:   "export deploymentConfig",
+			object: deploytest.OkDeploymentConfig(1),
+			expectedObj: &deployapi.DeploymentConfig{
+				ObjectMeta: kapi.ObjectMeta{
+					Name: "config",
+				},
+				LatestVersion: 0,
+				Triggers: []deployapi.DeploymentTriggerPolicy{
+					deploytest.OkImageChangeTrigger(),
+				},
+				Template: deploytest.OkDeploymentTemplate(),
+			},
+			expectedErr: nil,
+		},
+		{
+			name: "export imageStream",
+			object: &imageapi.ImageStream{
+				ObjectMeta: kapi.ObjectMeta{
+					Name:      "test",
+					Namespace: "other",
+				},
+				Spec: imageapi.ImageStreamSpec{
+					Tags: map[string]imageapi.TagReference{
+						"v1": {
+							Annotations: map[string]string{"an": "annotation"},
+						},
+					},
+				},
+				Status: imageapi.ImageStreamStatus{
+					DockerImageRepository: "foo/bar",
+					Tags: map[string]imageapi.TagEventList{
+						"v1": {
+							Items: []imageapi.TagEvent{{Image: "the image"}},
+						},
+					},
+				},
+			},
+			expectedObj: &imageapi.ImageStream{
+				ObjectMeta: kapi.ObjectMeta{
+					Name:      "test",
+					Namespace: "",
+				},
+				Spec: imageapi.ImageStreamSpec{
+					Tags: map[string]imageapi.TagReference{
+						"v1": {
+							From: &kapi.ObjectReference{
+								Kind: "DockerImage",
+								Name: "foo/bar:v1",
+							},
+							Annotations: map[string]string{"an": "annotation"},
+						},
+					},
+				},
+				Status: imageapi.ImageStreamStatus{
+					Tags: map[string]imageapi.TagEventList{},
+				},
+			},
+			expectedErr: nil,
+		},
+	}
+
+	for _, test := range tests {
+		if err := exporter.Export(test.object, test.exact); err != test.expectedErr {
+			t.Errorf("error mismatch: expected %v, got %v", test.expectedErr, err)
+		}
+
+		if !reflect.DeepEqual(test.object, test.expectedObj) {
+			t.Errorf("object mismatch: expected \n%v\ngot \n%v\n", test.expectedObj, test.object)
+		}
+	}
+}

